web_json_widget/static/src/components/json_editor/json_editor.esm.js

```diff
@@ -0,0 +1,133 @@
+// eslint-disable-next-line jsdoc/check-tag-names
+/** @odoo-module **/
+
+/* global JSONEditor */
+
+/*
+    Copyright 2025 Lambdao
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+*/
+
+import {
+    Component,
+    onMounted,
+    onPatched,
+    onWillUnmount,
+    useRef,
+    useState,
+} from "@odoo/owl";
+import {_t} from "@web/core/l10n/translation";
+import {cookie} from "@web/core/browser/cookie";
+import {registry} from "@web/core/registry";
+import {standardFieldProps} from "@web/views/fields/standard_field_props";
+
+export class JsonEditorComponent extends Component {
+    static template = "web_json_widget.JsonEditor";
+    static components = {};
+    static props = {
+        ...standardFieldProps,
+        // Placeholder: { type: String, optional: true },
+    };
+    static defaultProps = {};
+
+    setup() {
+        this.editor = null;
+        this.editorRef = useRef("editor");
+        this.resId = this.props.record.resId;
+        this.resModel = this.props.record.resModel;
+        this.state = useState({
+            value: this.props.record.data[this.props.name] || {},
+        });
+
+        onMounted(() => {
+            if (this.editorRef.el) {
+                this.createEditor();
+            }
+        });
+
+        onPatched(() => {
+            if (
+                this.resId !== this.props.record.resId ||
+                this.resModel !== this.props.record.resModel
+            ) {
+                this.resId = this.props.record.resId;
+                this.resModel = this.props.record.resModel;
+                if (this.editor) {
+                    this.editor.set(this.props.record.data[this.props.name] || {});
+                }
+            }
+        });
+
+        onWillUnmount(() => {
+            if (this.editor) {
+                this.editor.destroy();
+            }
+        });
+    }
+
+    createEditor() {
+        // Ensure the target element exists and is a DOM element
+        if (!this.editorRef.el || !(this.editorRef.el instanceof Element)) {
+            console.error("Target element not available for JSON editor");
+            return;
+        }
+        if (typeof JSONEditor === "undefined") {
+            console.error("JSONEditor not available");
+            return;
+        }
+
+        const isDarkMode = this.detect_dark_mode();
+        if (isDarkMode) {
+            this.editorRef.el.classList.add("json-editor-theme-dark");
```

Comment:
I wonder why you don't do this in the template?

Reply:
Not sure exactly but the doc about the Json editor was a bit confusing to me, and I found confusing answers as to how to instantiate a dark version. I think the 'Odoo way' is to put that in a CSS without the dark mangling which is put into the dark assets bundle. As of now I find it simpler to keep it this way because I copy-pasted this from recommended setup, but it wouldn't be hard to change afterwards.